pkg/operator/loglevel/controller_test.go

```diff
@@ -0,0 +1,115 @@
+package loglevel
+
+import (
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	logLevelUtils "github.com/openshift/library-go/pkg/operator/loglevel"
+
+	"github.com/openshift/cloud-credential-operator/pkg/operator/constants"
+	schemeutils "github.com/openshift/cloud-credential-operator/pkg/util"
+)
+
+const (
+	testCRName    = "cluster"
+	testNamespace = "openshift-cloud-credential-operator"
+)
+
+type ExpectedCondition struct {
+	klogLevel   operatorv1.LogLevel
+	logrusLevel log.Level
+}
+
+func TestLogLevelCloudCredentialReconcile(t *testing.T) {
+	schemeutils.SetupScheme(scheme.Scheme)
+
+	tests := []struct {
+		name               string
+		existing           []runtime.Object
+		expectedConditions ExpectedCondition
+	}{
+		{
+			name: "set debug log level",
+			existing: []runtime.Object{
+				testOperatorConfig(operatorv1.Debug, operatorv1.Debug),
+			},
+			expectedConditions: ExpectedCondition{
+				logrusLevel: log.DebugLevel,
+				klogLevel:   operatorv1.Debug,
+			},
+		}, {
+			name: "set normal log level",
+			existing: []runtime.Object{
+				testOperatorConfig(operatorv1.Normal, operatorv1.Normal),
+			},
+			expectedConditions: ExpectedCondition{
+				logrusLevel: log.InfoLevel,
+				klogLevel:   operatorv1.Normal,
+			},
+		}, {
+			name: "ensure max loglevel is chosen",
+			existing: []runtime.Object{
+				testOperatorConfig(operatorv1.TraceAll, operatorv1.Normal),
+			},
+			expectedConditions: ExpectedCondition{
+				logrusLevel: log.TraceLevel,
+				klogLevel:   operatorv1.TraceAll,
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fakeClient := fake.NewFakeClient(test.existing...)
+			rccc := &ReconcileCloudCredentialConfig{
+				Client: fakeClient,
+			}
+
+			_, err := rccc.Reconcile(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      constants.CloudCredOperatorConfig,
+					Namespace: testNamespace,
+				},
+			})
+			if err != nil {
+				require.NoError(t, err, "Unexpected error: %v", err)
+			}
```

Comment:
Don't really need to check whether err != nil, just require that it not be an error unconditionally
```suggestion
			require.NoError(t, err, "Unexpected error: %v", err)
```

Reply:
Thanks for the advice, super new to unit-testing in Go. Youtube++ 

All fixed up now. How do you normally merge here? (automatically squash commits or should I manually squash ?)

Reply:
Go ahead an squash on your end, and then we can lgtm it and put it in the merge queue

Reply:
Cool, I've squashed into a single commit as this seems to be what most PRs are. If you want it split into multiple, I can roll it back. Otherwise, hopefully this'll get me my first OpenShift lgtm 😄